hivemind/utils/asyncio.py

```diff
@@ -114,23 +114,30 @@ async def amap_in_executor(
     queue = asyncio.Queue(max_prefetch)
 
     async def _put_items():
-        async for args in azip(*iterables):
-            await queue.put(loop.run_in_executor(executor, func, *args))
-        await queue.put(None)
+        try:
+            async for args in azip(*iterables):
+                await queue.put(loop.run_in_executor(executor, func, *args))
+            await queue.put(None)
+        except BaseException as e:
+            await queue.put(e)
```

Comment:
Suggestion: let's do it like this and revert the change below. This will remove this edge case as well as unnecessary `isinstance()` check.

```suggestion
            future = asyncio.Future()
            future.set_exception(e)
            await queue.put(future)
```